### PR TITLE
Add support for "-override" and "-overerr"

### LINKED ISSUE #26: Add "Override" compiler options
A couple of new compiler options were added to support the new `Override` keyword.
It would be nice if MaxIDE could enable/disable these via the usual menus.

* `-override` : Requires `Override` to be used on any methods which override another. Raises a warning during compilation.
* `-overrer` : This turns the above warning into an error, halting the build.

### PR DESCRIPTION
**Pull request by GWRon**
Closes #26

### CODE REVIEW

**Comment by GWRon** [comment]
@woollybah 
Did I do it different than how you wanted it - or did you simply miss the PR?

**Comment by woollybah** [comment]
I see... the "review" was only pending... that's what happens when you click on things you never used before :-p

**Comment by GWRon** [comment]
I had this before but thought to better keep it the way the other settings work.

**Comment by GWRon** [comment]
If you already wrote the code...just reject this PR and commit your own stuff. No problem

**Comment by woollybah** [comment]
I haven't written anything yet. Was working on makemd changes recently.
I also haven't used this review thing before. Did you get 3 separate notes/mails about changes, that you are responding to? In the main thread you can respond to each individually. (or at least, that's what I can see)

**Comment by GWRon** [comment]
No i directly replied to the mail i've received. It contained the complete review in one piece.

File: maxide.bmx [hunk 1]
```diff
@@ -5396,6 +5398,9 @@ Type TOpenCode Extends TToolPanel
 			If universal cmd :+ " -i"
 			If warnover cmd :+ " -w"
 			If gdbdebug cmd :+ " -gdb"
+			If requireOverride cmd :+ " -override"
+			'bmk requires "-override" to use "-overerr"
+			If requireOverride and overrideError cmd :+ " -overerr"
```
[the remark this answers is not in the document]
Comment (GWRon) [reply maxide.bmx]: Just want to note that I find it odd to use `Then` for multiline and no "splitter" for single-line `if` constructs.

Would assume the `then` would help to distinguish between a "bracket less" function+param call and the "consequence" of the if clause.

File: default.language.ini [hunk 2]
```diff
@@ -282,6 +282,8 @@ menu_program_buildoptions_verbose            = "Verbose Build"
 menu_program_buildoptions_quickscan          = "Quick Scan"
 menu_program_buildoptions_universal          = "Universal Build"
 menu_program_buildoptions_warnover           = "Overload Warnings"
+menu_program_buildoptions_requireoverride    = "Require 'Override' declaration"
+menu_program_buildoptions_overrideerrors     = "Raise 'override' errors not warnings"
```
[the remark this answers is not in the document]
Comment (GWRon) [reply default.language.ini]: Yes, I could be "concise" as the other labels:

Require 'Override'
'Override' Errors

But I doubt they are that helpful :-)

**Comment by GWRon** [comment]
Regarding "3 separate notes" and replying by mail: you see that little "letter" icon next to the user name which indicates if one replied via email (you can just "reply" to a notification mail and it appears here - just do not forget to remove the "quoted reply" es it else will get insered here too - exposing some information about the email).


Will change the code above according to your suggestions - and will "disable" the "override error" entry (and variable=false) when the primer one gets disabled.

If you answer meanwhile we could also introduce a "cache": so it "saves" if "override error" was active before the "override required" was disabled. Enabling the latter again will also enable "override error" again. While it adds convenience it might be less "obvious" what happens there. Your thoughts?

File: maxide.bmx (hunk 1 above)
[the remark this answers is not in the document]
Comment (woollybah) [reply maxide.bmx]: I actually dislike the single line `If` statements, but don't see much point re-writing everything in preference for my own fanciful ideals :-p

**Comment by woollybah** [comment]
You can probably leave the error "flag" as-is, and just have it disabled (but checked?) in the menu.
Otherwise, if you toggle the "override" menu while the error menu is checked, you'd then have to re-apply the error setting.

**Comment by GWRon** [comment]
Added a new commit. Excuse the inline comments but I needed to explain why I did it that way - on some calls you need to make sure the other widget exists already - also "CheckMenu" seems to indirectly call something like "EnableMenu". This has effect on the order of commands - and that you eg cannot do it in a prior "if"-case already.

**Comment by GWRon** [comment]
Not sure if it was required but I set the "propsed changes" to "resolved". Feel free to merge - or propose other stuff to add.

**Comment by GWRon** [comment]
Bump ... merge or propse changes.

Think this feature is worth to get added before the next NG release is done.

**Pull request merged**